src/types/UiaTextRangeBase.cpp

```diff
@@ -990,1534 +903,359 @@ const COORD UiaTextRangeBase::_getScreenFontSize() const
 {
-    return pData->GetTextBuffer().TotalRowCount();
+    FAIL_FAST_IF(!(viewport.Bottom >= viewport.Top));
+    // + 1 because COORD is inclusive on both sides so subtracting top
+    // and bottom gets rid of 1 more then it should.
+    return viewport.Bottom - viewport.Top + 1;
 }
 
 // Routine Description:
-// - Gets the width of the screen buffer rows
+// - adds the relevant coordinate points from the row to coords.
+// - it is assumed that startAnchor and endAnchor are within the same row
 // Arguments:
-// - pData - the UiaData for the terminal we are operating on
+// - startAnchor - the start anchor of interested data within the viewport. In text buffer coordinate space. Inclusive.
+// - endAnchor - the end anchor of interested data within the viewport. In text buffer coordinate space. Inclusive
+// - coords - vector to add the calculated coords to
 // Return Value:
-// - The row width
-const unsigned int UiaTextRangeBase::_getRowWidth(gsl::not_null<IUiaData*> pData)
+// - <none>
+void UiaTextRangeBase::_getBoundingRect(_In_ const COORD startAnchor, _In_ const COORD endAnchor, _Inout_ std::vector<double>& coords) const
 {
-    // make sure that we can't leak a 0
-    return std::max(static_cast<unsigned int>(_getScreenBufferCoords(pData).X), 1u);
-}
+    FAIL_FAST_IF(startAnchor.Y != endAnchor.Y);
 
-// Routine Description:
-// - calculates the column refered to by the endpoint.
-// Arguments:
-// - pData - the UiaData for the terminal we are operating on
-// - endpoint - the endpoint to translate
-// Return Value:
-// - the column value
-const Column UiaTextRangeBase::_endpointToColumn(gsl::not_null<IUiaData*> pData, const Endpoint endpoint)
-{
-    return endpoint % _getRowWidth(pData);
-}
+    const auto viewport = _pData->GetViewport();
+    const auto currentFontSize = _getScreenFontSize();
 
-// Routine Description:
-// - converts an Endpoint into its equivalent text buffer row.
-// Arguments:
-// - pData - the UiaData for the terminal we are operating on
-// - endpoint - the endpoint to convert
-// Return Value:
-// - the text buffer row value
-const TextBufferRow UiaTextRangeBase::_endpointToTextBufferRow(gsl::not_null<IUiaData*> pData,
-                                                               const Endpoint endpoint)
-{
-    return endpoint / _getRowWidth(pData);
-}
+    POINT topLeft{ 0 };
+    POINT bottomRight{ 0 };
 
-// Routine Description:
-// - counts the number of rows that are fully or partially part of the
-// range.
-// Arguments:
-// - pData - the UiaData for the terminal we are operating on
-// Return Value:
-// - The number of rows in the range.
-const unsigned int UiaTextRangeBase::_rowCountInRange(gsl::not_null<IUiaData*> pData) const
-{
-    if (_degenerate)
+    // startAnchor is converted to the viewport coordinate space
+    auto startCoord = startAnchor;
+    viewport.ConvertToOrigin(&startCoord);
+    topLeft.x = startCoord.X * currentFontSize.X;

```

Comment:
I feel like @miniksa will have something to say about the right way to use safemath throughout this function

Reply:
We're going to hold off on a massive SafeMath refactor in this space. I've gone ahead and sprinkled a few in though.